canonical-web-and-design/maas-ui
# feat(ui): update warning messages for when no node devices are found

## Done

- Moved the warning portion of `NodeDevices` to its own component
- Added different warning messages depending on the state of the machine

## QA

### MAAS deployment

To run this branch you will need access to one of the following MAAS deployments:

- [Karura](/HACKING.md#karura)
- [Bolla](/HACKING.md#bolla)
- [A development MAAS](/HACKING.md#development-deployment)
- [A local snap MAAS](/HACKING.md#snap-deployment) (this will not usually have machines)

### Running the branch

You can run this branch by:

- Serving with [dotrun](/HACKING.md#maas-ui-development-setup)
- [Building in a development MAAS](/HACKING.md#running-maas-ui-from-a-development-maas)

### QA steps

- This is easiest to QA on a `make sampledata`d MAAS that has a lot of machines that haven't been commissioned recently.
- Go to the PCI tab of a machine in `Failed testing` state and check that the warning message says to override failed testing before commissioning
- Go to the PCI tab of a machine in `Deployed` state and check that the warning message says to release the machine before commissioning
- Go to the PCI tab of a locked machine and check that the warning message says to unlock the machine before commissioning
- Go to the PCI tab of a machine that can't be commissioned for any other reason (e.g. an action is currently taking place, like testing) and check that the warning message says that commissioning can't be run at this time

## Fixes

Fixes canonical-web-and-design/MAAS-squad#2313
